### ISSUE TITLE
[Bug]: Changing matploblibrc xtick.major.size and xtick.minor.size has no effect

### ISSUE THREAD

**Issue by willcerny** [matplotlib/matplotlib#27699]
### Bug summary

Changing the parameters
 xtick.major.size
xtick.minor.size 
xtick.major.width
xtick.minor.width

has no effect on the resulting plots. This is **not** true for the ytick equivalents, and changing xtick.direction works fine as well

### Code for reproduction

```python
Using the default matplotlibrc file and changing any of the above listed parameters has no effect on any figure created with plt.figure().
```

### Actual outcome

No effect is seen for any changes in  xtick.major.size , xtick.minor.size , xtick.major.width , xtick.minor.width

### Expected outcome

Changing these parameters should change the tick lengths.

### Matplotlib Version

3.8.2

### Python version

3.9.18

### Installation

pip

**Comment by ksunden** [comment]
I do not see this problem, note that the default rcParams file has all of the values actually commented out, so if you change it you need to remove the `# `

**Comment by willcerny** [comment]
Thanks for the response @ksunden . I am certain they aren't commented out - as noted above, changing other parameters in the file seems to work to work just fine. I'm attaching my file in case it offers any insight.

[matplotlibrc.txt](https://github.com/matplotlib/matplotlib/files/14054396/matplotlibrc.txt)

**Comment by ksunden** [comment]
With that `matplotlibrc`, I see everything working as documented. I am able to change xtick parameters and see their effect in the plots generated.

**Comment by willcerny** [comment]
Interesting! I have no idea why it isn't working on my end, then. This was a brand-new environment for Python3.9.18 and matplotlib3.8.2. Were you testing with the same versions?

**Comment by ksunden** [comment]
I was originally using py3.11 and `main`, but the relevant code paths have not been touched to my knowledge since 3.8.2 and shouldn't be python version dependent. To be sure, I did test on py3.9 and released 3.8.2, but still see everything working.

Are you using a non-agg based backend (e.g. cairo, svg, pdf)?

Do you see changes if you update at runtime? (e.g. `mpl.rcParams["xtick.major.size"] = 10`) Do you read the value you set in the rcparams file before setting it?

Could there be other things about your environment that are overwriting rcParams? (e.g. libraries that you import that set things (that would be generally ill advised, but could happen), a PYTHONSTARTUP script, etc)

**Comment by timhoffm** [comment]
Closing for lack of response. Please report back and answer above questions if this is still relevant.
